src/libraries/System.Private.CoreLib/src/System/Numerics/BFloat16.cs

```diff
@@ -0,0 +1,148 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Numerics
+{
+    /// <summary>
+    /// Represents a shortened (16-bit) version of 32 bit floating-point value (<see cref="float"/>).
+    /// </summary>
+    public readonly struct BFloat16
```

Comment:
Implement `IFloatingPointIeee754<BFloat16>` as well?

Reply:
Yep, we do want to expose the full API surface, same as was done for `Half`:
```csharp
        : IComparable,
          ISpanFormattable,
          IComparable<BFloat16>,
          IEquatable<BFloat16>,
          IBinaryFloatingPointIeee754<BFloat16>,
          IMinMaxValue<BFloat16>,
          IUtf8SpanFormattable,
          IBinaryFloatParseAndFormatInfo<BFloat16>
```

Reply:
I assume BFloat16 should be aware of the primitive types in System namespace, but they should not be aware of BFloat16.

How are the formatting traits (max precision) determined? Is there an easy way to calculate them?

Reply:
> I assume BFloat16 should be aware of the primitive types in System namespace, but they should not be aware of BFloat16.

Right. It should be much like `Half` which is a user-defined type, not an IL primitive. So it defines all the conversion APIs, etc.

> How are the formatting traits (max precision) determined? Is there an easy way to calculate them?

The maximum number of significant digits is `96` (so `Number.BFloat16NumberBufferLength` should be `98`).

The general algorithm is covered by https://www.exploringbinary.com/maximum-number-of-decimal-digits-in-binary-floating-point-numbers/, but you need to account for leading zeros, which isn't as simple to do.

In general if you simply set the `BiasedExponent` to `1` and all `TrailingSignificand` bits to `1`, then doing `ToString("G999")` will give you a string representation of the value from which you can get the most significant digits.

For example, the following bitwise values will hit the maximum significant digit count:
* `half` - `0x07FF`
* `bfloat16` - `0x00FF` (so `0x00FF_0000` as single)
* `single` - `0xFF_FFFF`
* `double` - `0x001F_FFFF_FFFF_FFFF`